examples/4337-passkeys/src/routes/DeploySafe.tsx

```diff
@@ -20,12 +21,42 @@ import {
 } from '../logic/userOp'
 import { useUserOpGasLimitEstimation } from '../hooks/useUserOpGasEstimation'
 import { RequestStatus } from '../utils'
-import { PrefundCard } from './OpPrefundCard'
+import { PrefundCard } from '../components/OpPrefundCard'
 import { useFeeData } from '../hooks/useFeeData'
 import { useNativeTokenBalance } from '../hooks/useNativeTokenBalance'
 import { useCodeAtAddress } from '../hooks/useCodeAtAddress'
+import { getSafeRoute, HOME } from './constants.ts'
+import { OutletContext } from '../types/Outlet.ts'
+
+const loader: LoaderFunction = async () => {
+  const passkey = getPasskeyFromLocalStorage()
+
+  if (!passkey) {
+    return redirect(HOME)
+  }
+
+  const initializer: SafeInitializer = {
+    singleton: SAFE_SINGLETON_ADDRESS,
+    fallbackHandler: SAFE_4337_MODULE_ADDRESS,
+    signerFactory: WEBAUTHN_SIGNER_FACTORY_ADDRESS,
+    signerX: passkey.pubkeyCoordinates.x,
+    signerY: passkey.pubkeyCoordinates.y,
+    signerVerifier: P256_VERIFIER_ADDRESS,
+    setupTo: SAFE_MODULE_SETUP_ADDRESS,
+    setupData: encodeSafeModuleSetupCall([SAFE_4337_MODULE_ADDRESS]),
+  }
+  const initHash = getInitHash(initializer, APP_CHAIN_ID)
+  const launchpadInitializer = getLaunchpadInitializer(initHash)
+  const safeAddress = getSafeAddress(launchpadInitializer)
+
+  return { passkey, safeAddress }
+}
+
+function DeploySafe() {
+  const { passkey, safeAddress } = useLoaderData() as { passkey: PasskeyLocalStorageFormat; safeAddress: string }
```

Comment:
Unfortunately, this doesn't work because the loader/useLoaderData doesn't accept any type arguments, and the return type of the loader function is either the data or a response type (redirect, for example)